#24: Add option for file association

wtf911 opened the pull request:
This will add a checkbox so after installation finishes there will be an option to associate .nzb files with NZBGet. It will also remove the association if NZBGet is uninstalled. This solves issue number #24.

wtf911 committed:
Add option for file association

This will add a checkbox so after installation finishes there will be an option to associate .nzb files with NZBGet. It will also remove the association if NZBGet is uninstalled.

hugbug commented:
Thank you.

Uninstaller deletes ".nzb" key unconditionally. Wouldn't it be better to delete it only if it is set to "NZBGet.NZBFile", so that if user has changed association to another program the uninstaller wouldn't delete it?

wtf911 commented:
Yes that would be better. :) I also still want to figure out how to have it open NZBGet if it isn't already running.

wtf911 committed:
only delete key if associated

this adds an if statement to only delete the registry key if nzbget is associated with .nzb files at the time of uninstall

hugbug commented:
> I also still want to figure out how to have it open NZBGet if it isn't already running.

Instead of starting with `nzbget.exe -A %1` use another (new) switch to let nzbget uniquely identify that it was started with double click, for example `nzbget.exe -add %1`.

In [`WinConsole::InitAppMode`](https://github.com/nzbget/nzbget/blob/develop/daemon/windows/WinConsole.cpp#L103) check that switch and check if nzbget is already running (see  [`WinConsole::CheckRunning`](https://github.com/nzbget/nzbget/blob/develop/daemon/windows/WinConsole.cpp#L681)).

- if nzbget isn't running - modify the command line to start as server in application mode (with tray icon), `InitAppMode` has an example of that. Then enqueue nzb-file with [`Scanner::AddExternalFile`](https://github.com/nzbget/nzbget/blob/develop/daemon/queue/Scanner.cpp#L482). You may need to wait until nzbget is fully initialized, something like:
    ```C++
	while (!DownloadQueue::IsLoaded())
	{
		usleep(20 * 1000);
	}
    ```
   The waiting and enqueuing of nzb must be done in `WinConsole::Run` which runs in a separate thread. Doing all that in `InitAppMode` would block the program start.
  Note: the modification of command line is necessary so that on reloading via webui it wouldn't try to enqueue file again (it fully reinitializes and parses command line again);

- if nzbget is already running - send the nzb-file to running instance using [`RemoteClient`](https://github.com/nzbget/nzbget/blob/develop/daemon/main/nzbget.cpp#L759). That's how `nzbget.exe -A` would do that. Then terminate.

hugbug commented:
> if nzbget isn't running - modify the command line to start as server in application mode 

Or maybe instead launch another (server) instance of nzbget and wait until it initializes, then proceed as in "if nzbget is already running".

wtf911 commented:
This goes over my head a bit haha. Learning to use NSIS and enough python to make the sorting script was one thing. Now I'm going to learn to use C++? :D Hmm. I can try but if I'm going to do it I might need more of a nudge in the right direction at some point.

hugbug commented:
>Now I'm going to learn to use C++?

I thought you know it ;)

I'm going to merge this PR in current state as it's already useful.

wtf911 commented:
Nope, I don't know C++. Or any programming language proficiently haha. And sadly it looks like I won't be able to figure out how to add a new switch and get this working. I think I need your help. Thank you for letting me be a part of this. :)
